Common/AnalysisTools.h

```diff
@@ -159,3 +159,55 @@ int FindMatching(const LorentzVectorM& target_p4, const RVecLV& ref_p4,const flo
   } 
   return current_idx;
 }
+
+RVecI FindMatching(const RVecLV& target_p4, const RVecLV& ref_p4,const float deltaR_thr){
+  RVecI targetIndices(target_p4.size(), -1); 
+  RVecI refIndicesTaken; 
+  for(int targetIdx =0; targetIdx<target_p4.size(); targetIdx++){
+    int refIdxFound = FindMatching(target_p4[targetIdx], ref_p4, deltaR_thr);
+    if(std::find(refIndicesTaken.begin(), refIndicesTaken.end(), refIdxFound)!=refIndicesTaken.end()){
+      continue;
+    }
+    targetIndices[targetIdx] = refIdxFound;
+    if(refIdxFound!=-1){
+      refIndicesTaken.push_back(refIdxFound);
+    }
+  } 
+  return targetIndices;
+}
+
+
+namespace v_ops{
+
+RVecF pt(const RVecLV& p4){
+    RVecF pt; 
+    for(int p4_idx=0;p4_idx<p4.size();++p4_idx){
+      pt.push_back(p4.at(p4_idx).pt());
```

Comment:
here and below:
- `push_back` is expensive - preallocate vector and then just set values.
- use template for RVecLV type